### PR TITLE
doc: add documentation for URLs

### PR DESCRIPTION
**Pull request by vangent**
My plan is to do something similar to what Eli did for "As" -- remove most of the documentation from the actual code and point to here instead, so we can document well once instead of copypasta everywhere.

### CODE REVIEW

File: doc.go [hunk 1]
```diff
@@ -37,6 +37,50 @@ For sample applications and a tutorial, see the samples directory
...
+
+In addition to creating portable types via provider-specific constructors
+(e.g., creating a blob.Bucket using s3blob.OpenBucket), many portable types
+can also be created using a URL. The scheme of the URL specifies the provider,
+and each provider implementation has code to convert the URL into the data
+needed to call its constructor. For example, calling blob.OpenBucket with
+"s3blob://my-bucket" will return a blob.Bucket created using s3blob.OpenBucket.
+
+Each portable API package will document the types that it supports opening
+by URL; for example, the "blob" package supports Buckets, while the "pubsub"
```
Comment (eliben) [inline doc.go]: Quotations are not consistent here. It's a shame godoc doesn't let us use backticks to format fixed-width-font stuff :-/

I think the convention in godoc.org is not putting quotes... Let's stick to that?

File: doc.go [hunk 2]
```diff
@@ -37,6 +37,50 @@ For sample applications and a tutorial, see the samples directory
...
+will export URLOpeners that give you more fine grained control over the
+arguments needed by the constructor. In particular, portable types opened via
+URL will often use default credentials from the environment. For example, the
+AWS URL openers use the credentials saved by "aws login" (we don't want to
+include credentials in the URL itself, since they are likely to be sensitive).
+
+  - Instantiate the provider's URLOpener with the specific fields you need.
+    For example, "s3blob.URLOpener{ConfigProvider: myAWSProvider}" using a
+    ConfigProvider that holds explicit AWS credentials.
+    - Create your own instance of the URLMux, e.g., "mymux := new(blob.URLMux)".
```
Comment (eliben) [inline doc.go]: misaligned bullet/hyphen?

File: doc.go (hunk 1 above)
Comment (vangent) [reply doc.go]: Done.

File: doc.go (hunk 2 above)
Comment (vangent) [reply doc.go]: Done.

**Pull request merged**